lib/merge.ts

```diff
@@ -1,8 +1,15 @@
 import mergeWith from 'lodash.mergewith';
 
 export const merge = mergeWith;
-export const mergeCustomizer = (_object: any, srcVal: any) => {
+export const mergeCustomizer = (
+  _object: any,
+  srcVal: any,
+  key: 'string',
+  object: any,
+) => {
   if (Array.isArray(srcVal)) {
     return srcVal;
+  } else if (srcVal === undefined && _object) {
```

Comment:
What is the intent behind the `&& _object` check here (bearing in mind that `_object` is actually the value of the original field)? Wouldn't that make it so `undefined` only overrides the initial object value if the original value was truthy:

```ts
merge({ foo: 'sdf' }, { foo: undefined }) // Correct: { foo: undefined }
merge({ foo: true }, { foo: undefined }) // Correct: { foo: undefined }
merge({ foo: false }, { foo: undefined }) // Incorrect: { foo: false }
merge({ foo: 0 }, { foo: undefined }) // Incorrect: { foo: 0 }
merge({ foo: null }, { foo: undefined }) // Incorrect: { foo: 0 }
```

I propose removing the `&& _object` check altogether. Could we also add some tests for at least one of those currently failing scenarios?

Reply:
You are right. It does miss those falsy cases when the object keys need to be merged as well. I have modified this behaviour in [commit](https://github.com/thoughtbot/fishery/pull/62/commits/fb875960b7098f7b56fe7c37bb774b098cb578cc) and also added 2 new tests for this behaviour.